kolibri/plugins/learn/assets/src/views/CustomChannelRenderer.vue

```diff
@@ -0,0 +1,202 @@
+<template>
+
+  <div>
+    <KContentRenderer
+      ref="customiframe"
+      class="content-renderer"
+      :kind="topic.kind"
+      :lang="topic.lang"
+      :files="topic.files"
+      :options="topic.options"
+    />
+    <!-- placeholder for jb's work on custom overlay -->
+    <KModal
+      v-if="open"
+      :title="currentContent.title"
+      cancelText="Close"
+      @cancel="open = false"
+    >
+      <KContentRenderer
+        ref="customiframe"
+        class="content-renderer"
+        :kind="currentContent.kind"
+        :lang="currentContent.lang"
+        :files="currentContent.files"
+        :options="currentContent.options"
+      />
+    </KModal>
+  </div>
+
+</template>
+
+
+<script>
+
+  import pick from 'lodash/pick';
+  import { mapState } from 'vuex';
+  import { ContentNodeResource } from 'kolibri.resources';
+  import Hashi from 'hashi';
+  import router from 'kolibri.coreVue.router';
+
+  export default {
+    name: 'CustomChannelRenderer',
+    data() {
+      return {
+        currentContent: this.topic,
+        open: false,
+      };
+    },
+    computed: {
+      ...mapState('topicsTree', ['topic']),
+      context() {
+        const urlParams = new URLSearchParams(window.location.search);
+        const fetchedEncodedContext = urlParams.has('context')
+          ? urlParams.get('context')
+          : { context: { customChannel: true } };
+        return decodeURI(fetchedEncodedContext);
+      },
+    },
+    mounted() {
+      this.hashi = new Hashi({ iframe: this.$refs.customiframe });
+      this.hashi.on('collectionrequested', message => {
+        this.fetchContentCollection(message);
+      });
+      this.hashi.on('modelrequested', message => {
+        this.fetchContentModel(message);
+      });
+      this.hashi.on('navigateto', message => {
+        this.navigateTo(message);
+      });
+      this.hashi.on('context', message => {
+        this.getOrUpdateContext(message);
+      });
+    },
+    methods: {
+      // helper functions for fetching data from kolibri
+      // called in mainClient.js
+
+      fetchContentCollection(message) {
+        const options = message.options;
+        const getParams = pick(options, ['ids', 'page', 'pageSize', 'parent']);
```

Comment:
Note that the backend does not currently support pagination, so should just pick `ids` and `parent` for now.